rust/protocol/src/storage/traits.rs

```diff
@@ -1,38 +1,72 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2022 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
+//! Traits defining several stores used throughout the Signal Protocol.
+
 use async_trait::async_trait;
 use uuid::Uuid;
 
-use crate::state::{PreKeyId, SignedPreKeyId};
 use crate::{
-    IdentityKey, IdentityKeyPair, PreKeyRecord, ProtocolAddress, Result, SenderKeyRecord,
-    SessionRecord, SignedPreKeyRecord,
+    address::ProtocolAddress,
+    error::Result,
+    sender_keys::SenderKeyRecord,
+    state::{PreKeyId, PreKeyRecord, SessionRecord, SignedPreKeyId, SignedPreKeyRecord},
+    IdentityKey, IdentityKeyPair,
 };
 
+/// Handle to FFI-provided context object.
+///
+/// This object is not manipulated in Rust, but is instead passed back to each FFI
+/// method invocation. This argument should just be [None] for all clients of the Rust-only API.
 pub type Context = Option<*mut std::ffi::c_void>;
 
+/// Each Signal message can be considered to have exactly two participants, a sender and receiver.
+///
+/// [IdentityKeyStore::is_trusted_identity] uses this to ensure the identity provided is configured
+/// for the appropriate role.
+///
+/// *TODO: consider moving this enum into [crate::protocol]?*
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub enum Direction {
+    /// We are in the context of sending a message.
     Sending,
+    /// We are in the context of receiving a message.
     Receiving,
 }
 
+/// Interface defining the identity store, which may be in-memory, on-disk, etc.
+///
+/// Signal clients usually use the identity store in a [TOFU] manner, but this is not required.
+///
+/// [TOFU]: https://en.wikipedia.org/wiki/Trust_on_first_use
 #[async_trait(?Send)]
 pub trait IdentityKeyStore {
+    /// Return the single specific identity the store is assumed to represent, with private key.
     async fn get_identity_key_pair(&self, ctx: Context) -> Result<IdentityKeyPair>;
 
+    /// Return a [u32] specific to this store instance.
+    ///
+    /// This local registration id is separate from the per-device identifier used in
+    /// [ProtocolAddress] and should not change run over run.
+    ///
+    /// If the same *device* is unregistered, then registers again, the [ProtocolAddress::device_id]
+    /// will be the same, but the store registration id returned by this method will be regenerated.
```

Comment:
"may be the same" and "should be regenerated", but otherwise well summarized. :-)

Reply:
Done! Thanks :D